packages/v4/patternfly-docs/content/design-guidelines/styles/icons/IconRecommendations.js

```diff
@@ -53,154 +51,142 @@ const iconSvgs = {
   'pficon-settings': pficonSettings
 };
 
-export class IconRecommendations extends React.Component {
-  state = {
-    searchValue: '',
-    columns: [
-      { title: 'Old icon', transforms: [sortable], props: { className: css(styles.modifiers.fitContent)} },
-      { title: 'Updated icon', transforms: [sortable], props: { className: css(styles.modifiers.fitContent)} },
-      'Contextual usage'
-    ],
-    sortBy: {}
-  };
+export const IconRecommendations = () => {
+  const [searchValue, setSearchValue] = React.useState('');
+  const [sortByIndex, setSortByIndex] = React.useState();
+  const [sortDirection, setSortDirection] = React.useState();
 
-  handleSearchChange = (event, _checked) => {
-    const searchValue = event.target.value;
-    this.setState(() => ({
-      searchValue
-    }));
-  };
+  const getSortParams = columnIndex => ({
+    sortBy: {
+      index: sortByIndex,
+      direction: sortDirection,
+      defaultDirection: 'asc' // starting sort direction when first sorting a column. Defaults to 'asc'
+    },
+    onSort: (_event, index, direction) => {
+      setSortByIndex(index);
+      setSortDirection(direction);
+    },
+    columnIndex
+  });
 
-  onSort = (_event, index, direction) => {
-    this.setState({
-      sortBy: {
-        index,
-        direction
-      }
+  const filteredRows = React.useMemo(() => {
+    return recommendationsArray.filter(recommendation => {
+      const oldRec = recommendation[0];
+      const newRec = recommendation[1];
+      return oldRec.iconName.toLowerCase().includes(searchValue.toLowerCase()) ||
+        oldRec.iconUsage.toLowerCase().includes(searchValue.toLowerCase()) ||
+        oldRec.type.toLowerCase().includes(searchValue.toLowerCase()) ||
+        newRec.iconName.toLowerCase().includes(searchValue.toLowerCase()) ||
+        newRec.iconUsage.toLowerCase().includes(searchValue.toLowerCase()) ||
+        newRec.type.toLowerCase().includes(searchValue.toLowerCase())
     });
-  }
+  }, [searchValue]);
```

Comment:
This is hard coded to change 2 results (oldRec & newRec), but some rows have up to 5 results (4 old icons, 1 new icon) so suggest refactoring for a more dynamic approach to cover all use cases.  Also incorporates a separate recursive function to pull text out from `children` for the case of `Spinner` which otherwise was breaking upon search due to passing nested JSX rather than a searchable string to the `iconUsage` field:

```
const concatChildText = (children, text = '') => {
  React.Children.forEach(children, (child) => {
    if (typeof child === 'string') {
      text += child;
    } else if (React.isValidElement(child)) {
      text = concatChildText(child.props.children, text);
    }
  });
  return text;
}
```

```
const filteredRows = React.useMemo(() => {
    return recommendationsArray.filter(recommendation => (
      recommendation.some(rec => {
        return Object.values(rec).some(val => {
          const cellText = concatChildText(val);
          return cellText.toLowerCase().includes(searchValue.toLowerCase());
        });
      })
    ))
  }, [searchValue]);
```